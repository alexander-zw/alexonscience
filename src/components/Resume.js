/**
 * This component contains my resume.
 *
 * This page is for anyone who came to learn about me personally, including
 * who I am, my skills and interests, and some of my projects that can't be
 * accessed directly on my website.
 */
import React from "react";
import { Link } from "react-router-dom";
import "../styles/index.css";
import "../styles/Resume.css";
import alexander_wu from "../images/alexander_wu.jpg";
import MetaTags from "./subcomponents/MetaTags";

function Resume() {
    const thesisLink = "https://www2.eecs.berkeley.edu/Pubs/TechRpts/2021/EECS-2021-102.html";

    // "contents" is not including heading.
    const sections = [
        {
            id: "education",
            title: "Education",
            contents: (
                <div>
                    <p>
                        Thesis:{" "}
                        <a href={thesisLink}>
                            Optimizations and Improvements to Cryptographic Libraries for zkSNARKs
                        </a>
                    </p>
                </div>
            ),
        },
        {
            id: "projects",
            title: "Projects",
            contents: (
                <div>
                    <p>
                        See my <a href="https://github.com/alexander-zw">GitHub</a> for more.
                    </p>
                    <p>
                        <strong>Guavabots</strong>
                        <br />
                        NP-complete graph problem with greedy algorithms and integer linear
                        programming, Python
                    </p>
                    <p>
                        <strong>Pintos</strong>
                        <br />
                        Operating system file storage with FFS, priority thread scheduling, and
                        process execution, C
                    </p>
                    <p>
                        <strong>Secure File Storage</strong>
                        <br />
                        Cryptographically encrypted and signed file storage and sharing system, Go
                    </p>
                    <p>
                        <strong>The Labyrinth</strong>
                        <br />
                        2D maze game with world generation and game mechanics, Java
                    </p>
                    <p>
                        <strong>BearMaps</strong>
                        <br />
                        Interactive navigation application similar to Google Maps, uses graphs and
                        A* search, Java
                    </p>
                    <p>
                        <strong>Stereo Vision</strong>
                        <br />
                        Detecting depth with two different images, quad tree compression, C and
                        RISC-V
                    </p>
                </div>
            ),
        },
    ];

    const tableOfContents = sections.map((sect, index) => (
        <li className="contents-li" key={index}>
            <a href={`#${sect.id}`} className="contents-item">
                {sect.title}
            </a>
        </li>
    ));

    const content = sections.map((sect, index) => (
        <section id={sect.id} className="section" key={index}>
            <h2>{sect.title}</h2>
            {sect.contents}
        </section>
    ));

    return (
        <div className="outer-container top-margin bottom-margin">
            <MetaTags path="/resume" />

            <nav className="contents-nav text-div">
                <ol className="contents-ol">{tableOfContents}</ol>
            </nav>

            <div className="text-div">
                <img className="photo" src={alexander_wu} alt={"Alexander Wu"} />

                <span className="name-title">Alexander Wu</span>

                <p>
                    Hello! I am not currently seeking a new position, but you can{" "}
                    <Link to="/contact">contact me</Link> for other matters. I{"'"}m a software
                    developer who expresses his love for science through my YouTube channel, among
                    other things. I{"'"}m passionate about education and enjoy doing volunteering
                    work. In my spare time, I read novels, work on projects, practice Yoga, and ride
                    my bike everywhere.
                </p>

                {content}
            </div>
        </div>
    );
}

export default Resume;
