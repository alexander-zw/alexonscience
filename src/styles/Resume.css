.outer-container {
    display: grid;
    grid-template-columns: 20% auto;
    grid-template-areas: "nav content";
}

.contents-nav {
    position: -webkit-sticky; /* Safari earlier versions. */
    position: sticky;
    top: 3.2em;
    align-self: start;
}

.contents-ol {
    list-style-type: none;
}

.contents-li {
    margin-top: 1em;
}

.contents-item {
    text-decoration: none;
    color: #636363;
}

.contents-item:hover {
    color: #008388;
}

.photo {
    width: 4em;
    height: 4em;
    border-radius: 2em;
    transform: translate(0, 0.9em);
    margin-right: 1.5em;
}

.name-title {
    padding-top: 0.2em;
    font-size: 3em;
    font-family: "Oleo Script", "Georgia", serif;
    background-image: linear-gradient(to right, #249141, #249c8a, #079cb3, #3464eb);
    -webkit-background-clip: text;
    -moz-background-clip: text;
    background-clip: text;
    color: transparent;
}

.section {
    /* This allows linking via table of contents to arrive at the right place. */
    padding-top: 1.5em;
}
