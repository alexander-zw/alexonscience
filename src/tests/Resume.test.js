/**
 * Test for the Resume.js component.
 */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Resume from '../components/Resume';

it('renders correctly', () => {
    // Resume has a Link, so we surround it with a router.
    const tree = renderer.create(<BrowserRouter><Resume /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});
