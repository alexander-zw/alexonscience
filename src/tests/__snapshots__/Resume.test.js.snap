// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="outer-container top-margin bottom-margin"
>
  <nav
    className="contents-nav text-div"
  >
    <ol
      className="contents-ol"
    >
      <li
        className="contents-li"
      >
        <a
          className="contents-item"
          href="#education"
        >
          Education
        </a>
      </li>
      <li
        className="contents-li"
      >
        <a
          className="contents-item"
          href="#skills"
        >
          Skills
        </a>
      </li>
      <li
        className="contents-li"
      >
        <a
          className="contents-item"
          href="#awards"
        >
          Awards
        </a>
      </li>
      <li
        className="contents-li"
      >
        <a
          className="contents-item"
          href="#experience"
        >
          Experience
        </a>
      </li>
      <li
        className="contents-li"
      >
        <a
          className="contents-item"
          href="#projects"
        >
          Projects
        </a>
      </li>
    </ol>
  </nav>
  <div
    className="text-div"
  >
    <img
      alt="Alexander Wu"
      className="photo"
      src="alexander_wu.jpg"
    />
    <span
      className="name-title"
    >
      Alexander Wu
    </span>
    <p>
      I
      '
      m a current Master
      '
      s student studying Computer Science. I express my love for science through my YouTube channel, among other things. I
      '
      m passionate about education and enjoy doing volunteering work. In my spare time, I read novels, work on personal projects, practice Yoga, and bike around the Bay.
    </p>
    <p>
      Email: alexwu68 [at] berkeley [dot] edu (see
       
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact Me
      </a>
      )
    </p>
    <section
      className="section"
      id="education"
    >
      <h2>
        Education
      </h2>
      <div>
        <p>
          <strong>
            University of California, Berkeley
          </strong>
          : 08/2020 - 05/2021
           
          <br />
          Master of Science, Electrical Engineering and Computer Science (EECS), GPA 4.0
        </p>
        <p>
          <strong>
            University of California, Berkeley
          </strong>
          : 08/2017 - 05/2020
           
          <br />
          Bachelor of Science, Electrical Engineering and Computer Science (EECS), GPA 4.0
        </p>
      </div>
    </section>
    <section
      className="section"
      id="skills"
    >
      <h2>
        Skills
      </h2>
      <ul>
        <li>
          <strong>
            Programming
          </strong>
          : Java, C++, Python, Ruby, JavaScript, Go, C, Rust, HTML/CSS, SQL, Matlab, Scheme, RISC-V
        </li>
        <li>
          <strong>
            Frameworks
          </strong>
          : React, React Native, Django, Unity
        </li>
        <li>
          <strong>
            Other CS
          </strong>
          : Machine Learning, Algorithms, Computer Security, Web/Mobile Development
        </li>
        <li>
          <strong>
            Languages
          </strong>
          : English (native), Mandarin (native)
        </li>
      </ul>
    </section>
    <section
      className="section"
      id="awards"
    >
      <h2>
        Awards
      </h2>
      <ul>
        <li>
          Mark D. Weiser Excellence in Computing Scholarship (2020)
        </li>
        <li>
          Outstanding Team (highest international award),
           
          <em>
            18th COMAP High School Mathematics Contest in Modeling
          </em>
           (2016)
        </li>
        <li>
          National First Prize,
          <em>
            30th China Adolescents Science and Technology InnovationContest
          </em>
           
          (2015)
        </li>
        <li>
          International 66th Place and 7th in northern China,
           
          <em>
            13th AAPT PhysicsBowl
          </em>
           (2014)
        </li>
        <li>
          Chou Pei-Yuan Science and Innovation Award (2015)
        </li>
        <li>
          Soong Ching-ling Invention Award (2015)
        </li>
        <li>
          National First Prize, etc, 
          <em>
            FIRST Tech Challenge
          </em>
           (2014-15)
        </li>
      </ul>
    </section>
    <section
      className="section"
      id="experience"
    >
      <h2>
        Experience
      </h2>
      <div>
        <p>
          <strong>
            SDE Intern
          </strong>
           at Amazon.com Inc, Summer 2020
          <ul>
            <li>
              <strong>
                Image Gallery View
              </strong>
              : Migrated Amazon shopping app immersive product image gallery from obsolete native iOS and Android code to SSNAP (Amazon internal React Native framework), fixing 5 bugs and improving user experience, robustness, readability, and maintainability; released changes to beta, with 10,000+ loads in 10 regions within a week
            </li>
          </ul>
        </p>
        <p>
          <strong>
            AI Developer Intern
          </strong>
           at Stottler Henke Associates Inc, Summer 2019
          <ul>
            <li>
              <strong>
                MIDAS Automated Resource Scheduler
              </strong>
              : Implemented features, fixed bugs, and wrote tests for AFSCN satellite scheduling software in Java (large project with 20+ team members)
            </li>
            <li>
              <strong>
                Satellite Scheduling Network
              </strong>
              : Implemented/experimented with bottleneck scheduling algorithm, added features, GUIs, and visualizations, tested effectiveness and runtime, etc in Java, C++, and Python
            </li>
            <li>
              <strong>
                Overhead Persistent Infrared
              </strong>
              : Investigated a library for features, usage, and dependencies to interface sensors
            </li>
          </ul>
        </p>
        <p>
          <strong>
            Website Developer
          </strong>
           for HKN (IEEE honor society), Computing Services, Fall 2018 - Spring 2021
          <ul>
            <li>
              Developed the official website and made it user-friendly and appealing with Django web development
            </li>
          </ul>
        </p>
        <p>
          <strong>
            Research Assistant
          </strong>
           in cryptography for UC Berkeley EECS, Spring 2020, advisor Alessandro Chiesa
          <ul>
            <li>
              Developed features in cryptographic libraries for zkSNARKs in C++ and Rust
            </li>
          </ul>
        </p>
        <p>
          <strong>
            Research Assistant
          </strong>
          , conducted experiments for Department of Architecture at UC Berkeley, Summer and Fall 2018
          <ul>
            <li>
              Created a program with Java so that the data analyzing process was reduced from 3 weeks to 1 day
            </li>
            <li>
              Built and measured the properties of 3D-printed agro-waste materials in flood conditions
            </li>
          </ul>
        </p>
        <p>
          <strong>
            CS Instructor
          </strong>
          , part-time job for UC Berkeley Engineering (CS Bootcamp), Summers and Winters 2017-18
          <ul>
            <li>
              Taught online course and created instructional videos, exercises, auto-graders, documents, etc
            </li>
          </ul>
        </p>
        <p>
          <strong>
            Undergraduate Student Instructor
          </strong>
          , Discrete Math and Probability Theory (CS 70), UC Berkeley, Fall 2019 and Spring 2020
          <ul>
            <li>
              Taught discussion sections, hosted office hours, proctored/graded exams, contributed to course material, etc
            </li>
          </ul>
        </p>
      </div>
    </section>
    <section
      className="section"
      id="projects"
    >
      <h2>
        Projects
      </h2>
      <div>
        <p>
          See my 
          <a
            href="https://github.com/alexander-zw"
          >
            GitHub
          </a>
           for more.
        </p>
        <p>
          <strong>
            Guavabots
          </strong>
          <br />
          NP-complete graph problem with greedy algorithms and integer linear programming, Python
        </p>
        <p>
          <strong>
            Pintos
          </strong>
          <br />
          Operating system file storage with FFS, priority thread scheduling, and process execution, C
        </p>
        <p>
          <strong>
            Secure File Storage
          </strong>
          <br />
          Cryptographically encrypted and signed file storage and sharing system, Go
        </p>
        <p>
          <strong>
            The Labyrinth
          </strong>
          <br />
          2D maze game with world generation and game mechanics, Java
        </p>
        <p>
          <strong>
            BearMaps
          </strong>
          <br />
          Map of Berkeley similar to Google Maps, uses graphs and A* search, Java
        </p>
        <p>
          <strong>
            Stereo Vision
          </strong>
          <br />
          Detecting depth with two different images, quad tree compression, C and RISC-V
        </p>
      </div>
    </section>
  </div>
</div>
`;
